src/main/java/com/jcraft/jsch/Session.java

```diff
@@ -883,9 +885,20 @@ private void checkHost(String chost, int port, KeyExchange kex) throws JSchExcep
       }
       // System.err.println("finger-print: "+key_fprint);
       if (userinfo != null) {
+		if (shkc.equals("ask")) {
         boolean foo = userinfo.promptYesNo("The authenticity of host '" + chost
             + "' can't be established.\n" + key_type + " key fingerprint is " + key_fprint + ".\n"
             + "Are you sure you want to continue connecting?");
+		} else {
```

Comment:
This appears to be a dead branch because up above we already throw a JSchException if `shkc.equals("yes")`.